CORE/world_gadgets/data_engine.rb

```diff
@@ -0,0 +1,47 @@
+
+
+class DataEngine
+
+  def initialize(logger=nil)
+    @logger = logger
+  end
+
+  def load_data_from_yml(filename)
+    @logger.debug "Loading data from file #{filename}" if @logger
+    begin
+      data = YAML.load_file(filename)
+      return data
+    rescue Psych::SyntaxError => e
+      raise "ERROR: Tally could not parse this YML file!\n"\
```

Comment:
good catch by the way, I'm sure this won't be the last time that Tally -> OZ conversion will come up